### PR TITLE
Mixfile, deps and docs fixes

### PR DESCRIPTION
**Pull request by lucacorti**
- Add () in Mixfile (fixes 1.4 warnings)
- Update/simplify doc dependencies
- Fix code block indentation in docs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,3 @@ erl_crash.dump
 *.ez
 .mix_tasks
 /doc
-.plts
```
Comment (nesQuick) [inline .gitignore]: why that?

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,7 @@
 language: elixir
 
 elixir:
-  - 1.3.2
+  - 1.2.2
```
Comment (nesQuick) [inline .travis.yml]: I do not want to downgrade

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,7 @@
...
-cache:
-  directories:
-    - .plts
-
-before_script:
-  - "[[ -f .plts/local.plt ]] || mix dialyzer.plt"
-
-script:
-  - mix test
-  - mix dialyzer --check_plt
```
Comment (nesQuick) [inline .travis.yml]: your removing lots of stuff here, why?

File: lib/hpack.ex [hunk 4]
```diff
@@ -3,26 +3,24 @@ defmodule HPack do
...
   """
 
+  @type header :: {String.t, String.t}
+
   use Bitwise
   alias HPack.Huffman
   alias HPack.Table
 
-  @type header :: {String.t, String.t}
-  @type header_block_fragment :: binary
```
Comment (nesQuick) [inline lib/hpack.ex]: why remove that?

File: lib/hpack.ex [hunk 5]
```diff
@@ -3,26 +3,24 @@ defmodule HPack do
...
 
-    iex> {:ok, ctx} = HPack.Table.start_link(1000)
-    iex> HPack.encode([{":method", "GET"}], ctx)
-    << 0b10000010 >>
+      iex> {:ok, ctx} = HPack.Table.start_link(1000)
+      iex> HPack.encode([{":method", "GET"}], ctx)
+      << 0b10000010 >>
 
   """
-  @spec encode([header], Table.t) :: header_block_fragment
```
Comment (nesQuick) [inline lib/hpack.ex]: why remove that?

File: lib/hpack.ex [hunk 6]
```diff
@@ -69,12 +67,12 @@ defmodule HPack do
...
-    iex> {:ok, ctx} = HPack.Table.start_link(1000)
-    iex> HPack.decode(<< 0x82 >>, ctx)
-    [{":method", "GET"}]
+      iex> {:ok, ctx} = HPack.Table.start_link(1000)
+      iex> HPack.decode(<< 0x82 >>, ctx)
+      [{":method", "GET"}]
 
   """
-  @spec decode(header_block_fragment, Table.t) :: [header]
+  @spec decode(String.t, pid) :: [header]
```
Comment (nesQuick) [inline lib/hpack.ex]: I'd like to keep the `header_block_fragment` type

File: lib/hpack.ex [hunk 7]
```diff
@@ -130,37 +128,37 @@ defmodule HPack do
...
-  # | H |     Name Length (7+)      |
-  # +---+---------------------------+
-  # |  Name String (Length octets)  |
-  # +---+---------------------------+
   # | H |     Value Length (7+)     |
   # +---+---------------------------+
   # | Value String (Length octets)  |
   # +-------------------------------+
-  # Figure 9: Literal Header Field without Indexing — New Name
-  defp parse(<< 0::4, 0::4, rest::binary >>, headers, table) do
```
Comment (nesQuick) [inline lib/hpack.ex]: why are you reordering the functions?

File: lib/hpack/huffman.ex [hunk 8]
```diff
@@ -14,7 +14,7 @@ defmodule HPack.Huffman do
       "hello"
 
   """
-  @spec decode(binary) :: String.t
+  @spec decode(String.t) :: String.t
```
Comment (nesQuick) [inline lib/hpack/huffman.ex]: This is a binary type, not a String, while a String is always a binary not every binary is a String.

File: lib/hpack/huffman.ex [hunk 9]
```diff
@@ -29,7 +29,7 @@ defmodule HPack.Huffman do
       iex> HPack.Huffman.encode("hello")
       <<0x9c, 0xb4, 0x50, 0x7f>>
   """
-  @spec encode(String.t) :: binary
+  @spec encode(String.t) :: String.t
```
Comment (nesQuick) [inline lib/hpack/huffman.ex]: same binary comment as above

File: lib/hpack/table.ex [hunk 10]
```diff
@@ -1,10 +1,4 @@
 defmodule HPack.Table do
-  @moduledoc """
```
Comment (nesQuick) [inline lib/hpack/table.ex]: Why are you remove all of this comments and type hints in this file?

**Comment by lucacorti** [comment]
Sorry I obviously messed something up. Please ignore this, I'll create a new PR.

**Pull request closed without merging**